libnavigation-ui/src/main/java/com/mapbox/navigation/ui/feedback/FeedbackDetailsFragment.kt

```diff
@@ -0,0 +1,498 @@
+package com.mapbox.navigation.ui.feedback
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.Color
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.MotionEvent
+import android.view.View
+import android.view.View.GONE
+import android.view.View.INVISIBLE
+import android.view.View.VISIBLE
+import android.view.ViewGroup
+import android.view.inputmethod.InputMethodManager
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetBehavior.BottomSheetCallback
+import com.google.android.material.bottomsheet.BottomSheetBehavior.STATE_COLLAPSED
+import com.google.android.material.bottomsheet.BottomSheetBehavior.STATE_HIDDEN
+import com.google.android.material.bottomsheet.BottomSheetBehavior.from
+import com.mapbox.navigation.core.telemetry.events.CachedNavigationFeedbackEvent
+import com.mapbox.navigation.core.telemetry.events.FeedbackEvent
+import com.mapbox.navigation.core.telemetry.events.FeedbackEvent.POSITIONING_ISSUE
+import com.mapbox.navigation.ui.R
+import com.mapbox.navigation.ui.internal.utils.ViewUtils
+import kotlinx.android.synthetic.main.mapbox_edit_text_feedback_optional_comment.*
+import kotlinx.android.synthetic.main.mapbox_feedback_details_bottom_sheet.*
+import kotlinx.android.synthetic.main.mapbox_feedback_details_fragment.*
+import kotlinx.android.synthetic.main.mapbox_partial_feedback_bottom_sheet_top_banner.*
+import kotlinx.android.synthetic.main.mapbox_partial_feedback_details_button_bar.*
+
+/**
+ * This [Fragment] is responsible for showing UI related to providing
+ * more specific information about reported feedback topics.
+ */
+class FeedbackDetailsFragment private constructor() : DialogFragment() {
```

Comment:
Could you extract the non-view-dependent business logic outside of the fragment context and add unit tests?

Reply:
I agree with adding unit tests but I don't really see too much pure business logic that can be extracted.